swarm-derive/tests/test.rs

```diff
@@ -43,8 +43,14 @@ fn one_field() {
     }
 
     #[allow(dead_code)]
+    #[allow(unreachable_code)]
     fn foo() {
-        require_net_behaviour::<Foo>();
+        let _out_event: <Foo as NetworkBehaviour>::OutEvent = unimplemented!();
+        match _out_event {
+            FooEvent::Ping(event) => {
+                let _: libp2p::ping::Event = event;
+            }
```

Comment:
Why not just extend the pattern match?

```suggestion
            FooEvent::Ping(libp2p::ping::Event { .. }) => { }
```

Reply:
Done in https://github.com/libp2p/rust-libp2p/pull/2792/commits/ff636810156a54fd9eb63566ed1093352ef650bb. Though unfortunately I don't see a way to do the same for the identify and kademlia event given that they are enums.